js/search.js

```diff
@@ -45,35 +45,53 @@ class SearchInput extends Component {
   }
 }
 
-const SingleResultView = r => {
+function resultUri(result) {
+  return (
+    "/d/" + result.group_name + "/" + result.jar_name + "/" + result.version
+  );
+}
+
+const SingleResultView = (r, idx, isSelected, onMouseOver) => {
   const project =
     r.group_name === r.jar_name
       ? r.group_name
       : r.group_name + "/" + r.jar_name;
-  const docsUri = "/d/" + r.group_name + "/" + r.jar_name + "/" + r.version;
+  const docsUri = resultUri(r);
   return h("a", { className: "no-underline black", href: docsUri }, [
-    h("div", { className: "pa3 bb b--light-gray" }, [
-      h("h4", { className: "dib ma0" }, [
-        project,
-        h("span", { className: "ml2 gray normal" }, r.version)
-      ]),
-      h(
-        "a",
-        {
-          className: "link blue ml2",
-          href: docsUri
-        },
-        "view docs"
-      )
-      // h('span', {}, r.created)
-    ])
+    h(
+      "div",
+      {
+        className: isSelected
+          ? "pa3 bb b--light-gray bg-light-blue"
+          : "pa3 bb b--light-gray",
+        onMouseOver: () => {
+          onMouseOver(idx);
+        }
+      },
+      [
+        h("h4", { className: "dib ma0" }, [
+          project,
+          h("span", { className: "ml2 gray normal" }, r.version)
+        ]),
+        h(
+          "a",
+          {
+            className: "link blue ml2",
+            href: docsUri
+          },
+          "view docs"
+        )
+        // h('span', {}, r.created)
+      ]
+    )
   ]);
 };
 
 const ResultsView = props => {
   return h(
     "div",
     {
+      id: "results-view",
```

Comment:
I'm wondering if it would make sense to use refs here? It seems like the `App` component now has some expectations on the `ResultsView` without that being visible/explicit in any way. This isn't a huge issue for now I guess, just wondering if there's another way?

Reply:
Maybe one way to do it would be to introduce a `didUpdate` hook to the `ResultsView`, calling `restrictToViewport` whenever the `selectedIndex` passed via `props` changes.

Reply:
No worries in that case, I can probably fix this myself.